Dynatrace/dynatrace-operator
# Add additional KSPM clusterrole

## Description

[DAQ-2079](https://dt-rnd.atlassian.net/browse/DAQ-2079)

1. Add additional KSPM cluster role
2. Rename KSPM related helm values
3. Add KSPM to `install-config`
4. Add validation for KSPM module presence

Sidenote: 
I tested the scenario when `rbac.activegate.create: false` and `rbac.kspm.create: true`, which is an incorrect config because `kspm` needs the `activegate`, however it "just works".
- Helm install will work, even though the new `kspm` clusterrole is bound to the service account deployed for `activegate`, even if the service account is not present
- The module validation will catch it, because you have to explicitly define the `spec.activeGate` section in the DynaKube for the `kspm` feature to work.
   - If you specify the `spec.activeGate` section AND turned of the `rbac` in the helm values, then the validation will catch it. 🥳 

## How can this be tested?

Check that the new helm value works, and the validation works

[DAQ-2079]: https://dt-rnd.atlassian.net/browse/DAQ-2079?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
